logstash-core/lib/logstash/pipeline.rb

```diff
@@ -310,15 +302,35 @@ def output_batch(batch)
 
       # output_func should never return anything other than an Array but we have lots of legacy specs
       # that monkeypatch it and return nil. We can deprecate  "|| []" after fixing these specs
-      outputs_for_event = output_func(event) || []
+      outputs_for_event = with_pipeline_rescue(event) { output_func(event) } || []
 
       outputs_for_event.each { |output| acc[output] << event }
       acc
     end
 
     # Now that we have our output to event mapping we can just invoke each output
     # once with its list of events
-    outputs_events.each { |output, events| output.multi_receive(events) }
+    outputs_events.each do |output, events|
+      with_pipeline_rescue(events) do
+        output.multi_receive(events)
+      end
+    end
+  end
+
+  def with_pipeline_rescue(event_or_events)
+    yield
+  rescue Exception => e
+    logger_opts = {:message => e.message, :class => e.class.name}
+    logger_message = "Exception in pipelineworker!"
+
+    if @logger.info?
+      logger_opts[:events] = Array(event_or_events).map(&:to_hash)
+    end
+    
+    logger_message << "Enable info logging to see failed events" if !@logger.info?
```

Comment:
Lets also bump some metrics that we actually continued on for this event. We can expose this to the user.